File: idscp2/src/main/java/de/fhg/aisec/ids/idscp2/InputListener.java
package de.fhg.aisec.ids.idscp2;

/**
 * A event source interface for DataAvailableListener pattern
 *
 * @author Leon Beckmann (leon.beckmann@aisec.fraunhofer.de)
 */
public interface InputListener {

    void register(DataAvailableListener listener);

    void unregister(DataAvailableListener listener);

    void safeStop();
}

File: idscp2/src/main/kotlin/de/fhg/aisec/ids/idscp2/default_drivers/secure_channel/client/InputListener.kt
package de.fhg.aisec.ids.idscp2.default_drivers.secure_channel.client

/**
 * An interface for an InputListenerThread, that allows DataAvailableListener registration and
 * safe stop
 *
 * @author Leon Beckmann (leon.beckmann@aisec.fraunhofer.de)
 */
interface InputListener {
    /*
     * Register a listener for providing data to
     */
    fun register(listener: DataAvailableListener)

    /*
     * Terminate InputListener
     */
    fun safeStop()
}